File: app/src/main/java/com/unis/gameplatfrom/BaseApplication.java
package com.unis.gameplatfrom;

import android.app.ActivityManager;
import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;
import android.util.Log;

import java.util.List;

/**
 * Created by wulei on 2016-08-29.
 */

@SuppressWarnings("DefaultFileTemplate")
public class BaseApplication extends Application {
    private static BaseApplication instance;

    public static BaseApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;

    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

    /**
     * 判断APP是否在前台显示
     *
     * @param context
     * @return
     */
    public static boolean isBackground(Context context) {
        ActivityManager activityManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        List<ActivityManager.RunningAppProcessInfo> appProcesses = activityManager.getRunningAppProcesses();
        for (ActivityManager.RunningAppProcessInfo appProcess : appProcesses) {
            if (appProcess.processName.equals(context.getPackageName())) {
                /*
                BACKGROUND=400 EMPTY=500 FOREGROUND=100
                GONE=1000 PERCEPTIBLE=130 SERVICE=300 ISIBLE=200
                 */
                Log.i("debug", "MApp   >>此appimportace =" + appProcess.importance
                        + ",context.getClass().getName()=" + context.getClass().getName());
                if (appProcess.importance != ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND) {
                    Log.i("debug", "MApp   >>处于后台" + appProcess.processName);
                    return true;
                } else {
                    Log.i("debug", "MApp   >>处于前台" + appProcess.processName);
                    return false;
                }
            }
        }
        return false;
    }
}

File: kotlin/src/main/java/com/unis/kotlin/BaseApplication.kt
package com.unis.kotlin

import android.app.Application
import android.content.Context
import com.blankj.utilcode.util.Utils
import org.litepal.LitePal

open class BaseApplication : Application(){



    companion object {
        private var instance: BaseApplication? = null
        private var context: Context? = null

        fun getInstance(): BaseApplication?{
            return instance
        }

        fun getContext(): Context? {
            return context
        }

    }

    override fun onCreate() {
        super.onCreate()
        instance = this

        //初始化工具类
        Utils.init(this)
        //sqlite组件的初始化
        LitePal.initialize(this)

        context = applicationContext

        //崩溃日志
        val handler = MyCrashHandler()
        Thread.setDefaultUncaughtExceptionHandler(handler)


    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)

    }


}


